src/main/java/org/jvnet/hudson/test/TailLog.java

```diff
@@ -0,0 +1,128 @@
+import java.io.UnsupportedEncodingException;
+import java.util.concurrent.Semaphore;
+import org.apache.commons.io.input.Tailer;
+import org.apache.commons.io.input.TailerListenerAdapter;
+import org.jvnet.hudson.test.recipes.LocalData;
+
+/**
+ * Utility to display the log of a build in real time.
+ * Unlike {@link BuildWatcher}, this works well with both {@link RealJenkinsRule} and {@link LocalData}.
+ * Use in a {@code try}-with-resources block, typically calling {@link #waitForCompletion} at the end.
+ */
+@SuppressFBWarnings(value = "PATH_TRAVERSAL_IN", justification = "irrelevant")
+public final class TailLog implements AutoCloseable {
+
+    private final Semaphore finished = new Semaphore(0);
+    private final Tailer tailer;
+    private final PrefixedOutputStream.Builder prefixedOutputStreamBuilder = PrefixedOutputStream.builder();
+
+    /**
+     * Watch a build already loaded in the current JVM.
+     */
+    public TailLog(Run<?, ?> b) {
+        this(b.getRootDir(), b.getParent().getFullName(), b.getNumber());
+    }
+
+    /**
+     * Watch a build expected to be loaded in the current JVM.
+     * <em>Note</em>: this constructor will not work for a branch project (child of {@code MultiBranchProject}).
+     * @param job a {@link Job#getFullName}
+     */
+    public TailLog(JenkinsRule jr, String job, int number) {
+        this(runRootDir(jr.jenkins.getRootDir(), job, number), job, number);
+    }
+
+    /**
+     * Watch a build expected to be loaded in a controller JVM.
+     * <em>Note</em>: this constructor will not work for a branch project (child of {@code MultiBranchProject}).
+     * @param job a {@link Job#getFullName}
+     */
+    public TailLog(RealJenkinsRule rjr, String job, int number) {
+        this(runRootDir(rjr.getHome(), job, number), job, number);
+    }
+
+    private static File runRootDir(File home, String job, int number) {
+        // For MultiBranchProject the last segment would be "branches" not "jobs":
+        return new File(home, "jobs/" + job.replace("/", "/jobs/") + "/builds/" + number);
+    }
+
+    /**
+     * Applies ANSI coloration to log lines produced by this instance.
+     * Ignored when on CI.
+     * Does not work when the build has already started by the time this method is called.
+     */
+    public TailLog withColor(PrefixedOutputStream.Color color) {
+        prefixedOutputStreamBuilder.withColor(color);
+        return this;
+    }
+
+    /**
+     * Watch a build expected to run at a specific file location.
+     * @param buildDirectory expected {@link Run#getRootDir}
+     * @param job a {@link Job#getFullName}
+     */
+    public TailLog(File buildDirectory, String job, int number) {
+        tailer = Tailer.create(new File(buildDirectory, "log"), new TailerListenerAdapter() {
+            PrintStream ps;
+            @Override
+            public void handle(String line) {
+                if (ps == null) {
+                    try { // TODO Java 11 use Charset overload
```

Comment:
This repo uses Java 11 so this can be solved now

Reply:
#528 :eyes: